docs/whats_new/v0.4.0.md

```diff
@@ -0,0 +1,152 @@
+re-introducing the more advanced HPO methods such as bayesopt in a future release.
+Additionally, removed 40,000 lines of out-dated code to streamline future development.
+@Innixma (#1397, #1411, #1414, #1431, #1443, #1511)
+- Added cloud training with AWS SageMaker tutorials. @gradientsky (#1375)
+- Added GPU training tutorial for tabular models. @gradientsky (#1527)
+- Added feature preprocessing tutorial for tabular. @willsmithorg (#1478)
+- Added `autogluon.common` to simplify dependency management for future submodules. @Innixma (#1386)
+- Removed `autogluon.mxnet` and `autogluon.extra` submodules as part of code cleanup. @Innixma (#1397, #1411, #1414)
+- Refactored logging to avoid interfering with other packages. @yinweisu (#1403)
+- Fixed logging output on Kaggle, previously no logs would be displayed while fitting AutoGluon in a Kaggle kernel. @Innixma (#1468)
+- Added platform tests for Linux, MacOS, and Windows. @yinweisu (#1464, #1506, #1513)
+- Added ROADMAP.md to highlight past, present, and future feature prioritization and progress to the community. @Innixma (#1420)
+- Various documentation and CI improvements
+  - @jwmueller (#1379, #1408, #1429)
+  - @gradientsky (#1383, #1387, #1471, #1500)
+  - @yinweisu (#1441, #1482, #1566, #1580)
+  - @willsmithorg (#1476, #1483)
+  - @Xilorole (#1526)
+  - @Innixma (#1452, #1453, #1528, #1577, #1584, #1588, #1593)
+- Various backend enhancements / refactoring / cleanup
+  - @DolanTheMFWizard (#1319)
+  - @gradientsky (#1320, #1366, #1385, #1448, #1488, #1490, #1570, #1576)
+  - @mseeger (#1349)
+  - @yinweisu (#1497, #1503, #1512, #1563, #1573)
+  - @willsmithorg (#1525, #1543)
+  - @Innixma (#1311, #1313, #1327, #1331, #1338, #1345, #1369, #1377, #1380, #1408, #1410, #1412, #1419, #1425, #1428, #1462, #1465, #1562, #1569, #1591, #1593)
+- Various bug fixes
+  - @jwmueller (#1314, #1356)
+  - @yinweisu (#1472, #1499, #1504, #1508, #1516)
+  - @gradientsky (#1514)
+  - @Innixma (#1304, #1325, #1326, #1337, #1365, #1395, #1405, #1587, #1599)
+
+## Text
+
+TODO
+
+## Tabular
+
+- Added `infer_limit` and `infer_limit_batch_size` as new fit-time constraints. This allows users to specify
+the desired end-to-end inference latency of the final model and AutoGluon will automatically train models
+to satisfy the constraint. This is extremely useful for online-inference scenarios where you need to satisfy an
+end-to-end latency constraint (for example 50ms). @Innixma (#1541, #1584)
+- Implemented parallel bag training via Ray. This results in a ~2x training speedup when bagging is enabled
+compared to v0.3.1 with the same hardware due to more efficient usage of resources
+for models that cannot effectively use all cores. @yinweisu (#1329, #1415, #1417, #1423)
+- Implemented automated semi-supervised and transductive learning in TabularPredictor.
+Try it out via `TabularPredictor.fit_pseudolabel(...)`! @DolanTheMFWizard (#1323, #1382)
+- Implemented automated feature pruning in TabularPredictor.
```

Comment:
may want to say `automated feature pruning (i.e. feature selection) in ....`

as some folks will be more familiar with the latter terminology and may think 'pruning' refers to neuron pruning in NN models.

Reply:
Good point! Updated